scipy/stats/tests/test_stats.py

```diff
@@ -5708,3 +5708,25 @@ def test_dist_perm(self):
                                                                random_state=1)
         assert_approx_equal(stat_dist, 0.163, significant=1)
         assert_approx_equal(pvalue_dist, 0.001, significant=1)
+
+class TestConfInt(object):
+    """ Test the computation of non-parametric
+    confidence intervals for quantiles
+    """
+    X = array([2, 8, 3, 6, 4, 1, 5, 9, 7], float)
```

Comment:
If these tests are from data this is published somewhere, so that we can confirm that these are the correct result, please list the reference.
If these tests don't confirm that `confint_quantile` is doing the right thing - just that it continues to do the _same_ thing - then I would ask for stronger tests. Is there anything you can compare against?

Also, please test your input validation. Can you think of any edge cases that might make sense to test that aren't specifically checked by input validation?

I would also prefer to see additional sanity checks - like, for a large number of samples, does the confidence interval get tighter? So could we do a test on, say, `np.linspace(0, 1, N)` for some arbitrary quantile and large `N` and observe that the upper and lower bound are close to that value?

Reply:
Regarding the first point, I have not found something doing the exact same thing. This was discussed a bit above: 
> > I just browsed through the R packages and found QuantileNPCI which seems to be doing about the same thing, although it seems it computes only two-sided CI (not 100% sure yet, I just had a quick look).
> > Unfortunately I am not proficient in R, so it's not easy for me to quickly compare the two implementations. But I can try computing the CI presented in the QuantileNPCI example; that will already do for a quick check.
> 
> I checked the example in the vignette of the [QuantileNPCI](https://cran.r-project.org/web/packages/QuantileNPCI/index.html) R module. The computation method is similar, but there are some differences in the precise bound definition:
> 
> * QuantileNPCI derives the fraction of the sample index that defines the bound with the desired level of confidence, which are computed by interpolation from the sample values
> * Our implementation is more conservative: we take as bound the first sample value that is guaranteed to give an CI with the desired level of confidence.
> 
> So in this context, I am not sure if it makes sense to write a benchmark using that R function (since they do not do exactly the same computation). In any case, I added the reference to the package in the function docstring.

> > If I understand the comment correctly, then the results should be the same when the fraction corresponds to an integer, i.e. no interpolation between two indices is needed.
(I have this case sometimes when writing unit tests for statsmodels, where I just compare cases that fall on integer values.)
> 
> I think so. 

Do you think it would be worthwhile to include such a test? 

Besides that, I can definitely add some more validation tests (that's the first time I wrote such code, I was not too sure how far I needed to go...). I will work on this and expand the test set.

Reply:
OK. Sometimes what I do, if possible is write a minimalistic implementation of the method right in the test suite (e.g. [here](https://github.com/scipy/scipy/pull/13330/files#diff-ecd325a8f3e4d3163bc96e89de3b8c9ec6b89f327a423f75f0cc7f13ed034ae9)). If you can boil it down to just a few lines that are easier to confirm (and if you write them on a different day, without copying and pasting so as not to be influenced by any possible mistakes above), it serves a bit like an independent check.

Reply:
Yes, that should be fine. I do have a different implementation that I can use for this. 

It was changed following earlier comments to leverage methods from the `binom` distribution. Anyhow, I see what I should do :-)